rosidl_runtime_c/include/rosidl_runtime_c/message_type_support_struct.h

```diff
@@ -29,6 +31,12 @@ typedef struct rosidl_message_type_support_t rosidl_message_type_support_t;
 typedef const rosidl_message_type_support_t * (* rosidl_message_typesupport_handle_function)(
   const rosidl_message_type_support_t *, const char *);
 
+typedef const rosidl_runtime_c__type_description__TypeDescription *
```

Comment:
I discovered a slight complication here when dealing with construction of `rosidl_message_type_support_t` structs at runtime.

Consider:
```
    new rosidl_message_type_support_t{
    rmw_dynamic_typesupport_c__identifier,                // typesupport_identifier
    ts_impl,                                              // data
    get_message_typesupport_handle_function,              // func
    type_hash.get(),                                      // type_hash
    XXX,                                                  // get_type_description_func
    XXX                                                   // get_type_description_sources_func
  }
```

In this case, the description will have been constructed and provided **at runtime**. There's no way to define a function at compile time that can return the type description. (A similar problem exists for sources, but there are potentially no sources in the runtime constructed case.)

Is it possible to change the signatures of the funcs to take in an optional pointer to the typesupport struct so that it becomes possible to do something like...

```
const rosidl_runtime_c__type_description__TypeDescription * my_dynamic_get_type_description_func(rosidl_message_type_support_t * ts) {
  if (ts->typesupport_identifier == rmw_dynamic_typesupport_c__identifier) {
    return (rmw_dynamic_message_typesupport_impl_t *)(ts->data)->description;
  }
}
```

Reply:
Ah yeah, you're right, no way to form a proper closure in C (at least, not one that takes no arguments). I guess I will have to change the signature. See followup message below about the hashes as well